src/jit/target.h

```diff
@@ -2299,65 +1993,6 @@ extern const regMaskTP raRbmCalleeSaveOrder[CNT_CALLEE_SAVED];
 // This method takes a "compact" bitset of the callee-saved registers, and "expands" it to a full register mask.
 regMaskSmall genRegMaskFromCalleeSavedMask(unsigned short);
 
-/*****************************************************************************
- *
- *  Returns the register that holds the low  32 bits of the long value given
- *  by the register pair 'regPair'.
- */
-inline regNumber genRegPairLo(regPairNo regPair)
```

Comment:
As I see logic for register pairs were changed. There were a lot issues related to it in ARM/ARM64.
Could you point where is new logic for long types?

Reply:
There are no reg pairs anymore; long decomposition for arm/x86 forces LONG to INT types in RyuJIT.

Reply:
As I remember pairs were needed to store long values in neighboring registers which is declared in ARM ABI. Does the new implementation work well with this rule?

Reply:
Yes. Note that we have been running with arm RyuJIT for a long time, with no failures. I do remember seeing code that uses regNum and REG_NEXT(regNum) instead of the register pairs, which might be what you are referring to. Note that the RegPair concept allowed any pair of registers, and was mostly used by the old register allocator. The new allocator, LSRA, requires long decomposition. LclVars still have lvOtherReg (for 32-bit targets) and lvOtherArgReg.

Reply:
Got it! Thank you

Reply:
@alpencolt - there are also a small number of nodes that support multiple registers, where there are actual target instructions that produce multiple registers (whereas decomposition expands instruction sequences when no such single instruction exists). Those instructions are: `GT_CALL`, `GT_COPY`, and `GT_RELOAD` (most targets), plus `GT_MUL_LONG`, `GT_PUTARG_REG`, `GT_PUTARG_SPLIT` and `GT_BITCAST` on ARM. Instead of using the `gtRegPair` these use additional register fields that are defined only for these nodes.
Note that this support has evolved a bit over time, and could use some cleanup (I think there may be an issue associated with that), though I believe that it is complete, just a but messier that it could be.